### ISSUE TITLE
Platform table filter - not properly used aria-selected

### ISSUE THREAD

**Issue by esporysz** [SAP/fundamental-ngx#12784]
#### Is this a bug, enhancement, or feature request?

bug

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)

angular: 18.2.4
fundamental: 0.53.1-rc.23

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;

1. Crete some platform table with sorting and filtering. It may be copy of [example from documentation](https://sap.github.io/fundamental-ngx/#/platform/table/preserved-state).
2. Click filter icon.
3. We can see filter dialog with 3 options 'price', 'status' and 'status color'. Each of them has a `role='listitem'` and `aria-selected =false`. 

According to [documentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected)  `aria-selected` should be used with `gridcell`, `option`, `row`, `tab` not `listitem` role.

**Comment by mikerodonnell89** [comment]
I believe @esporysz is referring to this dialog:
![Image](https://github.com/user-attachments/assets/6fcda6a7-a42a-4a9e-9d76-0682de03b6f8)
The obvious quick fix here would be to just use inputs to manually assign the correct roles `<ul fd-list role="listbox">` and `<li fd-list-item ariaRole="option">` in `filters-list-step.component.html`. But I am wondering if this list should have `[selection]="true"`, and if the listbox and option roles should be set automatically whenever `[selection]="true"` is set by the dev. Currently, the selection input is only used if the list items have radios or checkboxes. So using `[selection]="true"` on this list does break some styling. Thoughts @InnaAtanasova @droshev? I guess to summarize, my question is, do we consider this list to have "selection"?

**Comment by mikerodonnell89** [comment]
@esporysz you are referring to the list in this dialog, correct? Just want to make sure, I'm removing the `aria-selected` attribute from this list now

![Image](https://github.com/user-attachments/assets/e1d56e45-56c4-4625-97ee-54cad6bcf7eb)

**Comment by mikerodonnell89** [comment]
This is actually a problem with the `fd-list` component, where `aria-selected` is used when I think `aria-current="page"` is probably better for most examples at least on the core list docs page. But sometimes `aria-selected` would be better if the list item has a radio or checkbox. So I think we need to add a new input for handling `aria-current`, still applying the `is-selected` class if `aria-current` is used, and leaving it in the developers' hands to choose between current vs. selected

**Comment by mikerodonnell89** [comment]
@esporysz I believe you are referring to this dialog?

![Image](https://github.com/user-attachments/assets/607c113c-5e3c-43dc-9f2e-feb229ad07de)

This is actually a problem with the `fd-list` component, I believe we'll need to do a large scale refactor of the approach to the list and list items regarding accessibility that will likely be a breaking change for some users.
